transitions/index.js

```diff
@@ -140,7 +140,11 @@ const loadTransitions = (autoEnableSystemTransitions = true) => {
 const loadTransition = key => {
   try {
     logger.info(`loading transition ${key}`);
-    transitions[key] = require('./' + key);
+    const transition = require('./' + key);
+    if (transition.init) {
```

Comment:
I know this is the existing behaviour, but now might be a good time to take a step back and re-consider our approach in allowing Sentinel to run with misconfigured transitions.

IMO we should not be doing this, because there is a good chance people are not going to be following the logs closely enough to realise that a transition of theirs is not running. I realise they will probably test this the first time they run it, but I imagine they will not re-test on subsequent updates.

It's unfortunate that we don't have a reliable way to exposing these sorts of problems in sentinel like we do in api (ie we can just refuse to boot if we have issues there). 

Interested in your thoughts here, and also perhaps @henokgetachew's as well. I wonder how much effort it would be to quickly whip up a status endpoint for sentinel, and expose that to whatever monitoring software we have.

I don't want to pile on extra work solving this one problem. I can imagine the monitoring could be a second ticket. I am interested in how you feel about having Sentinel fail to boot on bad config as a part of this PR though.